### PR TITLE
Study groups page

### LINKED ISSUE #143: Create Pages for Student Login and Registration
Study Groups List and Individual Page
- [x] Fix the JS build bug and upload the requested fixes at [PR #140](https://github.com/w3b3d3v/web3-bootcamp-platform/pull/140).
- [x] Rebase with new multi-language features.

### PR DESCRIPTION
**Pull request by nomadbitcoin**
## Brief Information

The type of this pull request is:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

This PR introduces new features related to the management of study groups. The changes include:

1. **New Pages**:
   - A new page to list all study groups, allowing users to easily see all available options.
   - A new page dedicated to loading and displaying specific content for each study group.

2. **Component Enhancements**:
   - Modification of the existing `Tab` component to now load and display students who are part of courses and groups.

3. **Helper Functions**:
   - Addition of several to facilitate the addition of new courses or groups within the Firebase database. 

### Implementation Details

All the changes were made by reusing the existing logic and components of the codebase.

Closes #143

### CODE REVIEW

File: lib/course.js [hunk 1]
```diff
@@ -33,12 +33,44 @@ async function getCourseLessons(course) {
...
+  const docSnap = querySnapshot.docs[0];
+  return { id: docSnap.id, ...docSnap.data() };
+}
+
+const kindToFunctionMap = {
+  course: getCourseById,
+  group: getGroupBySlug
+};
+
+export async function getCourse(kind, identifier) {
```
Comment (danicuki) [inline lib/course.js:68]: we should rename this. `Study Groups` are not courses. I understand you intention, reusing code, but this signature confuses me. Also the implementation doesn't need this `kindToFunctionMap`. If can directly get the function from the kind. My suggestion:

```javascript
function getCourse(courseId) { }
function getStudyGroup(slug) { }

function getDocument(collection, id) { 
   return self["get" + collection](id)
} 

getDocument("StudyGroup", "DAO")
getDocument("Course", "Solidity")

File: pages/study-groups/[slug].js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            )}
+          </>
+        )}
+      </div>
+    </>
+  )
+}
+
+export async function getServerSideProps({ params }) {
+  const group = await getCourse("group", params.slug)
```
Comment (danicuki) [inline pages/study-groups/[slug].js:146]: lets call it "study group" and not group ;-)

File: pages/study-groups/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import Head from 'next/head'
+import { useState } from 'react'
+import { getAllGroups } from '../../lib/study-groups'
+import { Container } from '@nextui-org/react'
+import { GroupCard } from '../../components/Card/Group'
+import Image from 'next/image'
+ 
+function Groups({ AllGroups }) {
+  const pageInfo = {
+    "title": "List of All Groups",
```
Comment (danicuki) [inline pages/study-groups/index.js:10]: List of all Study Groups

File: pages/study-groups/index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import Head from 'next/head'
+import { useState } from 'react'
+import { getAllGroups } from '../../lib/study-groups'
+import { Container } from '@nextui-org/react'
+import { GroupCard } from '../../components/Card/Group'
+import Image from 'next/image'
+ 
+function Groups({ AllGroups }) {
```
Comment (danicuki) [inline pages/study-groups/index.js:8]: `StudyGroups`

**Review by danicuki — CHANGES_REQUESTED** [review]

File: components/Tabs/index.js [hunk 5]
```diff
@@ -3,6 +3,37 @@ import useAuth from '../../hooks/useAuth'
 import { checkSections, colorTab } from './tabFunctions'
 
 export default function Tabs({ course, isLessonPage, lessonsSubmitted, cohort }) {
```
[the remark this answers is not in the document]
Comment (nomadbitcoin) [reply components/Tabs/index.js:5]: I'm not sure if I got it.

Courses have an analytics structure with more data, including a list of objects with sessions, the number of students in each session and profile pictures.

Study groups will only have the number of students and their profile pictures. 

do you suggest that I just check that lessonsSubmitted and cohort are undefined and if they are assume that it is a study group?

**Comment by danicuki** [comment]
There are a lot of things still missing here. I will add them in a comment and accept the PR. @nomadbitcoin please work on these other things in a new PR. 

- translations i18n missing. It must use the new i18n translation files and add all keys there. We also need to have description and course details, title, etc, all this in the database, one for each language.
- When subscribe to group, the page that shows up is related to builds (live, etc). It must explain about the study group 
- It is not recording in the database that this user is subscribed to the group. It must record the same way we record cohorts / courses
- getStudyGroup function is in `courses.js` file, it should be in another place

**Pull request merged**